test/rules/no_duplicate_cases.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N no_adjacent_strings_in_list`
+
+void switchInt() {
+  const int A = 1;
+  int v = 5;
+
+  switch (v) {
+    case 1: // LINT
+    case 2: // LINT
+    case A: // LINT
+    case 2: // LINT
+    case 3: // OK
+    default:
+  }
+}
+
+void switchString() {
+  const String A = 'a';
+  String v = 'aa';
+
+  switch (v) {
+    case 'aa':  // LINT
+    case 'bb':  // LINT
+    case A + A: // LINT
+    case 'bb':  // LINT
+    case A + 'b': // OK
+    default:
+  }
```

Comment:
It would be good to test an enum type, and also a user defined type with a const constructor. I suspect they will work, but it's good to be complete.

Also, try testing something for which the ConstantVisitor will produce an error.